ledger/participant-local-store/src/test/lib/com/daml/platform/localstore/IdentityProviderConfigStoreTests.scala

```diff
@@ -0,0 +1,271 @@
+    }
+
+    "allows to update existing identity provider config's jwksUrl attribute" in {
+      testIt { tested =>
+        val cfg = config().copy(jwksURL = JwksUrl.assertFromString("http://daml.com/jwks1.json"))
+        for {
+          _ <- tested.createIdentityProviderConfig(cfg)
+          res2 <- tested.updateIdentityProviderConfig(
+            IdentityProviderConfigUpdate(
+              identityProviderId = cfg.identityProviderId,
+              jwksUrlUpdate = Some(JwksUrl.assertFromString("http://daml.com/jwks2.json")),
+            )
+          )
+        } yield {
+          res2 shouldBe Right(
+            cfg.copy(jwksURL = JwksUrl.assertFromString("http://daml.com/jwks2.json"))
+          )
+        }
+      }
+    }
+
+    "allows to update existing identity provider config's issuer attribute" in {
+      testIt { tested =>
+        val cfg = config().copy(issuer = "issuer1")
+        for {
+          _ <- tested.createIdentityProviderConfig(cfg)
+          res2 <- tested.updateIdentityProviderConfig(
+            IdentityProviderConfigUpdate(
+              identityProviderId = cfg.identityProviderId,
+              issuerUpdate = Some("issuer2"),
+            )
+          )
+        } yield {
+          res2 shouldBe Right(cfg.copy(issuer = "issuer2"))
+        }
+      }
+    }
+
+    "allows to update everything at the same time" in {
+      testIt { tested =>
+        val id = randomId()
+        val cfg = IdentityProviderConfig(
+          identityProviderId = id,
+          isDeactivated = false,
+          jwksURL = JwksUrl.assertFromString("http://example.com/jwks.json"),
+          issuer = UUID.randomUUID().toString,
+        )
+        for {
+          _ <- tested.createIdentityProviderConfig(cfg)
+          res <- tested.updateIdentityProviderConfig(
+            IdentityProviderConfigUpdate(
+              identityProviderId = cfg.identityProviderId,
+              issuerUpdate = Some("issuer2"),
+              jwksUrlUpdate = Some(JwksUrl.assertFromString("http://daml.com/jwks2.json")),
+              isDeactivatedUpdate = Some(true),
+            )
+          )
+        } yield {
+          res shouldBe Right(
+            IdentityProviderConfig(
+              identityProviderId = id,
+              isDeactivated = true,
+              jwksURL = JwksUrl.assertFromString("http://daml.com/jwks2.json"),
+              issuer = "issuer2",
+            )
+          )
+        }
+      }
+    }
+
+    "disallow updating issuer to non-unique value" in {
+      testIt { tested =>
+        val cfg1 = config().copy(issuer = "issuer1")
+        val cfg2 = config().copy(issuer = "issuer2")
+        for {
+          _ <- tested.createIdentityProviderConfig(cfg1)
+          _ <- tested.createIdentityProviderConfig(cfg2)
+          res <- tested.updateIdentityProviderConfig(
+            IdentityProviderConfigUpdate(
+              identityProviderId = cfg1.identityProviderId,
+              issuerUpdate = Some("issuer2"),
```

Comment:
What if you're doing a no-op update like this:
```
IdentityProviderConfigUpdate(
              identityProviderId = cfg1.identityProviderId,
              issuerUpdate = Some("issuer1"),
            )
```
Will an error still be raised?

Reply:
it is currently allowed, not sure what the expectation, but let me add a test case for that

Reply:
actually it is covered by the test and we do not expect failure in this case